requirements.txt

```diff
@@ -3,6 +3,7 @@ myst-parser
 furo
 sphinx-copybutton
 sphinx-autobuild
+pre-commit
```

Comment:
We don't need to install pre-commit to build the docs.

```suggestion
```

Instead, let's add a `.github/workflows/lint.yml`:

```yml
name: Lint

on: [push, pull_request, workflow_dispatch]

permissions:
  contents: read

jobs:
  lint:
    runs-on: ubuntu-latest

    steps:
      - uses: actions/checkout@v4
      - uses: actions/setup-python@v4
        with:
          python-version: "3.x"
      - uses: pre-commit/action@v3.0.0
```